### PR TITLE
[Broker] Make PersistentTopicsBase#terminate async

### PR DESCRIPTION
**Pull request by RobertIndie**
Master Issue: #14013

### Motivation
See #14013

### Modifications

* Make PersistentTopicsBase#terminate async

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -3289,23 +3289,38 @@ private RetentionPolicies getRetentionPolicies(TopicName topicName, TopicPolicie
 
     }
 
-    protected MessageId internalTerminate(boolean authoritative) {
+    protected void internalTerminateAsync(AsyncResponse asyncResponse, boolean authoritative) {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3292]: We'd better make this method return `CompletableFuture<MessageId>`, it's hard to get the response type for the rest API, only can be found here https://github.com/apache/pulsar/pull/14064/files#diff-66aeb65a64cbe7c541f013ae807c5bcbeab567bef77706c7ff2e0cbfe0d77eb1R3317

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java [hunk 2]
```diff
@@ -2675,7 +2676,14 @@ public MessageId terminate(
             @ApiParam(value = "Is authentication required to perform this operation")
             @QueryParam("authoritative") @DefaultValue("false") boolean authoritative) {
         validatePersistentTopicName(tenant, namespace, encodedTopic);
```
Comment (liudezhi2098) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java:2678]: need catch exception

Comment (RobertIndie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java:2678]: I think we don't need to. The exception it throws is of type RestException.

**Comment by RobertIndie** [comment]
These changes have been moved to https://github.com/apache/pulsar/pull/16227 . So close this PR.

**Pull request closed without merging**